identity/src/main/java/com/shotcutter/identity/user/UserIdentityService.java

```diff
@@ -37,4 +51,39 @@ public UserIdentityService(UserRepository userRepository,
                 .map(userRepository::save);
     }
 
+    public Optional<UserEntity> updateAvatar(String userId, MultipartFile avatar) throws IOException {
+        var path = getUserAvatarPath(userId, avatar);
+        var user = findById(userId).get();
+
+        // if avatar for this user already exist at s3 - we need to remove it at first
+        if (doesCurrentUserHaveAvatar(user)) {
+            var currentAvatarPath = new URL(user.getAvatar()).getPath();
+            amazonS3.deleteObject(s3BucketName, currentAvatarPath.substring(1));
+        }
+
+        var request = new PutObjectRequest(s3BucketName, path, avatar.getInputStream(), new ObjectMetadata())
+                                        .withCannedAcl(CannedAccessControlList.PublicRead);
+        amazonS3.putObject(request);
+        var amazonUrl = amazonS3.getUrl(s3BucketName, path);
+
+        var updatedUser = user.withAvatar(amazonUrl.toString());
+        return Optional.ofNullable(userRepository.save(updatedUser));
+    }
+
+    public Optional<UserEntity> patch(String userId, UserPatchDTO patchObject) {
+        return findById(userId)
+                .map(user -> user.withUsername(patchObject.getUsername()))
+                .map(userRepository::save);
+    }
+
+    private boolean doesCurrentUserHaveAvatar(UserEntity user) {
+        return user.getAvatar().contains(s3BucketName) && user.getAvatar().contains(user.getId());
```

Comment:
it doesn't really check if the picture stored in S3, please test the case when this method returns 'true' but there's nothing to delete in S3. I couldn't find any description for such case in docs fast, so quicker to check manually